src/main/java/no/unit/nva/bare/BareConnection.java

```diff
@@ -78,63 +73,75 @@ protected URI generateGetUrl(String systemControlNumber)
         URI uri = new URIBuilder()
                 .setScheme(HTTPS)
                 .setHost(Config.getInstance().getBareHost())
-                .setPath(Config.BARE_GET_PATH + "/" + systemControlNumber)
+                .setPath(Config.BARE_GET_PATH + PATH_SEPARATOR + systemControlNumber)
                 .setParameter("format", "json")
                 .build();
         return uri;
     }
 
     /**
      * Get an authority from Bare by given systemControlNumber.
+     *
      * @param systemControlNumber scn
      * @return InputStreamReader containing the authority payload
-     * @throws IOException some communication mishap
+     * @throws IOException        some communication mishap
      * @throws URISyntaxException error in configuration
+     * @throws InterruptedException error in communication
      */
-    public BareAuthority get(String systemControlNumber) throws URISyntaxException, IOException {
+    public BareAuthority get(String systemControlNumber) throws URISyntaxException, IOException, InterruptedException {
         final URI getUrl = generateGetUrl(systemControlNumber);
         System.out.println("bareConnection.get(" + getUrl + ")");
-        HttpGet httpGet = new HttpGet(getUrl);
-        try (CloseableHttpResponse response = httpClient.execute(httpGet)) {
-            if (response.getStatusLine().getStatusCode() == Response.Status.OK.getStatusCode()) {
-                try (Reader streamReader = new InputStreamReader(response.getEntity().getContent())) {
-                    BareAuthority fetchedAuthority = new Gson().fromJson(streamReader, BareAuthority.class);
-                    return fetchedAuthority;
-                }
-            } else {
-                System.out.println("Error..? " + response.getStatusLine().getReasonPhrase());
-                throw new IOException(response.getStatusLine().getReasonPhrase());
-            }
+        HttpRequest request = HttpRequest.newBuilder()
+                .GET()
+                .version(HttpClient.Version.HTTP_2)
+                .uri(getUrl)
+                .timeout(TIMEOUT_DURATION)
+                .build();
+        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
```

Comment:
Something like:

```
httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString())
                .thenApply(HttpResponse::body)
                .thenApply(body -> GSON.fromJson(body), BareAuthority.class)
                .exceptionally(IOException::new)
                .get();
```

Reply:
That is a valid suggestion. It leads to a different responsibilties in the classes.
I think taht is out of scope of that java 11 upgrading pull request.